src/org/rust/lang/codestyle/RustLanguageCodeStyleSettingsProvider.kt

```diff
@@ -0,0 +1,69 @@
+package org.rust.lang.codestyle
+
+import com.intellij.psi.codeStyle.CodeStyleSettingsCustomizable
+import com.intellij.psi.codeStyle.CommonCodeStyleSettings
+import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider
+import org.rust.lang.RustLanguage
+
+class RustLanguageCodeStyleSettingsProvider : LanguageCodeStyleSettingsProvider() {
+    override fun getLanguage() = RustLanguage.INSTANCE
+
+    override fun customizeSettings(consumer: CodeStyleSettingsCustomizable,
+                                   settingsType: LanguageCodeStyleSettingsProvider.SettingsType) {
+
+        if (settingsType == SettingsType.WRAPPING_AND_BRACES_SETTINGS) {
+            consumer.showStandardOptions("RIGHT_MARGIN");
+        }
+    }
+
+    override fun getDefaultCommonSettings() = CommonCodeStyleSettings(language).apply {
+        RIGHT_MARGIN = 99
+    }
+
+    override fun getCodeSample(settingsType: LanguageCodeStyleSettingsProvider.SettingsType) = "// A unit struct\n" +
+            "struct Nil;\n" +
```

Comment:
It would be better to extract the code sample into separate file and load it using ```StreamUtil``` see https://github.com/alexeykudinkin/intellij-rust/blob/2ccb70e75aa30c9a202c8eecd2c42afa586602ff/src/org/rust/lang/colorscheme/RustColorSettingsPage.kt#L33